File: modules/x-engine-jsi-legacy/iOS/Class/nav/JSIOldNavModule.m
//
//  JSIOldNavModule.h
//  ModuleApp
//
//  Created by zk on 2021/3/14.
//  Copyright © 2021 zkty-team. All rights reserved.
//

#import "JSIOldNavModule.h"
#import "JSIContext.h"
#import "XENativeContext.h"
#import "Unity.h"
#import "RecyleWebViewController.h"
#import "WebViewFactory.h"
#import "GlobalState.h"
#import "HistoryModel.h"
#import "iDirectManager.h"
#import "iUI.h"


@interface JSIOldNavModule ()
    @property (nonatomic, strong)   id<iDirectManager>  directors;
    @property (nonatomic, strong)   id<iUI>  ui;
    
@end

@implementation JSIOldNavModule
JSI_MODULE(JSIOldNavModule)

- (NSString*) moduleId{

    return @"com.zkty.module.nav";
}

-(void)afterAllJSIModuleInited {
    self.directors = [[XENativeContext sharedInstance] getModuleByProtocol:@protocol(iDirectManager)];
    self.ui = [[XENativeContext sharedInstance] getModuleByProtocol:@protocol(iUI)];
}


- (void)_navigatorBack:(NavNavigatorBackDTO *)dto complete:(void (^)(BOOL))completionHandler {
    [self.directors back:@"microapp" host:nil fragment:dto.url];
    if(completionHandler){
        completionHandler(YES);
    }
}

- (void)_navigatorPush:(NavNavigatorDTO *)dto complete:(void (^)(BOOL))completionHandler {

    NSString* scheme = @"microapp";
    /// TODO: convert dto.params to dictionary
    [self.directors push:scheme host:nil pathname:@"" fragment:dto.url query:nil params:@{@"hideNavbar":[NSNumber numberWithBool:dto.hideNavbar] }];
    if(completionHandler){
        completionHandler(YES);
    }}


-(void)_setNavBarHidden:(NavHiddenBarDTO2 *)dto complete:(void (^)(BOOL))completionHandler{
    [self.ui setNavBarHidden:dto.isHidden isAnimation:dto.isAnimation];
    completionHandler(YES);
}


- (void)_setNavTitle:(NavTitleDTO2 *)dto complete:(void (^)(BOOL))completionHandler {

    [self.ui setNavTitle:dto.title withTitleColor:dto.titleColor withTitleSize:dto.titleSize];
    completionHandler(YES);
}



@end

File: modules/x-engine-jsi-legacy/android/jsi-legacy/src/main/java/com/zkty/jsi/legacy/nav/JSIOldNavModule.java
package com.zkty.jsi.legacy.nav;

import android.webkit.JavascriptInterface;

import com.alibaba.fastjson.JSONObject;
import com.zkty.engine.nativ.protocol.IDirectManager;
import com.zkty.nativ.core.NativeContext;
import com.zkty.nativ.core.NativeModule;
import com.zkty.nativ.direct.NativeDirect;
import com.zkty.nativ.jsi.JSIModule;
import com.zkty.nativ.jsi.bridge.CompletionHandler;

import java.util.HashMap;
import java.util.Map;

public class JSIOldNavModule extends JSIModule {

    private NativeDirect directors;

    @Override
    public String moduleId() {
        return "com.zkty.module.nav";
    }

    @Override
    protected void afterAllJSIModuleInited() {

        NativeModule module = NativeContext.sharedInstance().getModuleByProtocol(IDirectManager.class);
        if (module instanceof NativeDirect)
            directors = (NativeDirect) module;
    }

    @JavascriptInterface
    public void navigatorPush(JSONObject obj, final CompletionHandler<Object> handler) {
        String scheme = "microapp";
        if (directors != null) {
            Map<String, Object> map = new HashMap<>();
            map.put("hideNavbar", obj.get("hideNavbar"));
            directors.push(scheme, null, null,
                    obj.getString("url"), null, map);
        }
    }

    @JavascriptInterface
    public void navigatorBack(JSONObject obj, final CompletionHandler<Object> handler) {
        String scheme = "microapp";
        if (directors != null) {
            directors.back(scheme, null, obj.getString("url"));
        }

    }


}
